src/pymorize/cmorizer.py

```diff
@@ -257,6 +260,25 @@ def _post_init_populate_rules_with_data_request_variables(self):
             self._rules_expand_drvs()
             self._rules_depluralize_drvs()
 
+    def _post_init_create_controlled_vocabularies(self):
+        """
+        Reads the controlled vocabularies from the directory tree rooted at
+        `<tables_dir>/CMIP6_CVs` and stores them in the `controlled_vocabularies`
+        attribute. This is done after the rules have been populated with the
+        tables and data request variables, which may be used to lookup the
+        controlled vocabularies.
+        """
+        table_dir = self._general_cfg["CMIP_Tables_Dir"]
+        cv_dir = Path(table_dir) / "../CMIP6_CVs"
+        self.controlled_vocabularies = ControlledVocabularies.new_from_dir(cv_dir)
```

Comment:
This is CMIP6-specific, I'll try to generalize it, and test it today or tomorrow.

Reply:
Sorted